File: frontend/src/models/ngo/donorLead.ts
import type { DonorLeadSource, DonorLeadStatus } from '@/src/constants/ngo/donorLead';

export type { DonorLeadSource, DonorLeadStatus };

export interface DonorLead {
  id: string;
  tenant_id: string;
  full_name: string;
  email: string;
  phone?: string | null;
  organization?: string | null;
  expected_donation: number;
  status: DonorLeadStatus;
  source: DonorLeadSource;
  assigned_to?: string | null;
  notes?: string | null;
  is_active: boolean;
  createdAt?: string | null;
  updatedAt?: string | null;
}

export interface DonorLeadCreate {
  full_name: string;
  email: string;
  phone?: string;
  organization?: string;
  expected_donation?: number;
  status?: DonorLeadStatus;
  source?: DonorLeadSource;
  assigned_to?: string;
  notes?: string;
}

export interface DonorLeadUpdate {
  full_name?: string;
  email?: string;
  phone?: string;
  organization?: string;
  expected_donation?: number;
  status?: DonorLeadStatus;
  source?: DonorLeadSource;
  assigned_to?: string;
  notes?: string;
  is_active?: boolean;
}

export interface DonorLeadsResponse {
  leads: DonorLead[];
  total: number;
}

export interface DonorLeadFilters {
  search?: string;
  status?: string;
  source?: string;
  created_date?: string;
}

File: backend/src/models/ngo/donor_lead.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Index, Numeric, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from ...config.database_config import Base


class DonorLead(Base):
    __tablename__ = "ngo_donor_leads"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, index=True)
    tenant_id = Column(UUID(as_uuid=True), ForeignKey("tenants.id", ondelete="CASCADE"), nullable=False)
    full_name = Column(String(255), nullable=False)
    email = Column(String(255), nullable=False)
    phone = Column(String(50))
    organization = Column(String(255))
    expected_donation = Column(Numeric(14, 2), nullable=False, default=0)
    status = Column(String(32), nullable=False, default="new")
    source = Column(String(32), nullable=False, default="other")
    assigned_to = Column(String(255))
    notes = Column(Text)
    is_active = Column(Boolean, default=True)
    createdAt = Column(DateTime, default=datetime.utcnow)
    updatedAt = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    __table_args__ = (
        Index("ix_ngo_donor_leads_tenant_id", "tenant_id"),
        Index("ix_ngo_donor_leads_tenant_status", "tenant_id", "status"),
    )

    tenant = relationship("Tenant")
